### ISSUE TITLE
gpm install admin not work in windows7

### ISSUE THREAD

**Issue by ericfjl** [getgrav/grav#1354]
`-------------------------commond--------------------`
`D:\phpStudy\WWW\OpenSource\CMS\grav>php bin/gpm install admin`
-------------------------result--------------------

> GPM Releases Configuration: Testing
> 
> 
> 
> Preparing to install Admin Panel [v1.3.0-rc.1]
>   |- Downloading package...   100%
> PHP Warning:  file_put_contents(D:/phpStudy/WWW/OpenSource/CMS/grav/tmp/Grav-58c
> a75a31ab04/admin1.3.0-rc.1?testing): failed to open stream: No such file or dire
> ctory in D:\phpStudy\WWW\OpenSource\CMS\grav\system\src\Grav\Console\Gpm\Install
> Command.php on line 599
> 
> Warning: file_put_contents(D:/phpStudy/WWW/OpenSource/CMS/grav/tmp/Grav-58ca75a3
> 1ab04/admin1.3.0-rc.1?testing): failed to open stream: No such file or directory
>  in D:\phpStudy\WWW\OpenSource\CMS\grav\system\src\Grav\Console\Gpm\InstallComma
> nd.php on line 599
>   |- Checking destination...  ok
>   |- Installing package...    error
>   |  '- An error occurred while extracting the package
>   '- Installation failed or aborted.
> 
> 
> Clearing cache
>

**Comment by OleVik** [comment]
This appears to be a problem with PHP not being able to write the file. Is `D:/phpStudy/WWW/OpenSource/CMS/grav` under any restrictions by Windows in terms of permissions? Generally speaking, the `SYSTEM` and `Administrators` group should have `Full Control`, as should the primary user.

**Comment by OleVik** [comment]
Actually, this is likely not an error unique to your system, but stems from how Grav handles pre-releases. I can replicate it on my otherwise fully functional Grav system, using v1.2.0-rc.2 and Admin v1.3.0-rc.2 (installed from downloaded package), where `php bin/gpm install adminstyles` will throw essentially the same error:

```
PHP Warning: 
file_put_contents(C:/Caddy/temp/tmp/Grav-58ce6cec0a160/adminstyles1.0.1?testing): 
failed to open stream: No such file or directory in 
C:\Caddy\temp\system\src\Grav\Console\Gpm\InstallCommand.php on line 599
```

This is seemingly because you were trying to install the latest Admin-plugin, and I the AdminStyles-plugin, both of which are currently in pre-release on GitHub. The error is not explicitly Windows permissions for the /tmp folder, as even with `Full Access` given to `Authenticated Users`, `SYSTEM`, `Administrators` and `Users` it persists. Rather, it looks like PHP will not actually create the folders then the files, thus failing the process.

I'm going to tag @rhukster and @w00fz, as they might have come over similar reports and can tag this issue as a "bug" or "investigating".

**Comment by OleVik** [comment]
This is ostensibly because Windows does not take kindly to `?` in filenames. Verified by the following:

`touch adminstyles-1.0.1?testing` creates a `adminstyles-1.0.1testing`.

`file_put_contents('adminstyles-1.0.1?testing.zip', file_get_contents('https://getgrav.org/download/plugins/adminstyles/1.0.1?testing'));` will fail with a `PHP Warning: file_put_contents...`.

`file_put_contents('adminstyles-1.0.1-testing.zip', file_get_contents('https://getgrav.org/download/plugins/adminstyles/1.0.1?testing'));` will work.

Thus the retrieved `Download: https://getgrav.org/download/plugins/adminstyles/1.0.1?testing` from GPM must either be rewritten to exclude `?`, or use a more common path that Windows will not reject.

**Comment by OleVik** [comment]
Solved with next release, see #1361.
